video/out/drm_atomic.c

```diff
@@ -452,3 +453,140 @@ bool drm_mode_destroy_blob(int fd, struct drm_mode *mode)
+    }
+
+    if(p->data.hdmi_metadata_type1.eotf) {
+        // Convert to nits (cd/m2):
+        float min_luminance = 0.0;
+        float max_luminance = color->sig_peak * MP_REF_WHITE;
+        float max_cll = max_luminance;
+        float max_fall = max_luminance;
+        
+        // Convert to 16-bit values in units of 0.00002, where 0x0000 represents zero and 0xC350 (50000) represents 1.0000
+        p->data.hdmi_metadata_type1.display_primaries[0].x = clamp16(roundf(colors.red.x * 50000.0));
+        p->data.hdmi_metadata_type1.display_primaries[0].y = clamp16(roundf(colors.red.y * 50000.0));
+        p->data.hdmi_metadata_type1.display_primaries[1].x = clamp16(roundf(colors.green.x * 50000.0));
+        p->data.hdmi_metadata_type1.display_primaries[1].y = clamp16(roundf(colors.green.y * 50000.0));
+        p->data.hdmi_metadata_type1.display_primaries[2].x = clamp16(roundf(colors.blue.x * 50000.0));
+        p->data.hdmi_metadata_type1.display_primaries[2].y = clamp16(roundf(colors.blue.y * 50000.0));
+        // White point
+        p->data.hdmi_metadata_type1.white_point.x = clamp16(roundf(colors.white.x * 50000.0));
+        p->data.hdmi_metadata_type1.white_point.y = clamp16(roundf(colors.white.y * 50000.0));
+
+        // Min Luminance: 16-bit value in units of 0.0001 cd/m2, where 0x0001 represents 0.0001 cd/m2 and 0xFFFF represents 6.5535 cd/m2.
+        p->data.hdmi_metadata_type1.min_display_mastering_luminance = clamp16(floorf(min_luminance * 10000.0));
+
+        // Max Luminance: 16-bit value in units of 1 cd/m2, where 0x0001 represents 1 cd/m2 and 0xFFFF represents 65535 cd/m2.
+        p->data.hdmi_metadata_type1.max_display_mastering_luminance = clamp16(ceilf(max_luminance));
+
+        // Max Content Light Level: 16-bit value in units of 1 cd/m2, where 0x0001 represents 1 cd/m2 and 0xFFFF represents 65535 cd/m2.
+        p->data.hdmi_metadata_type1.max_cll = clamp16(ceilf(max_cll));
+
+        // Max Frame Average Light Level: 16-bit value in units of 1 cd/m2, where 0x0001 represents 1 cd/m2 and 0xFFFF represents 65535 cd/m2.
+        p->data.hdmi_metadata_type1.max_fall = clamp16(ceilf(max_fall));
+
+        /*
+        printf("HDR: peak=%f min=%d max=%d cll=%d fall=%d\n",
+            color->sig_peak,
+            p->data.hdmi_metadata_type1.min_display_mastering_luminance,
+            p->data.hdmi_metadata_type1.max_display_mastering_luminance,
+            p->data.hdmi_metadata_type1.max_cll,
+            p->data.hdmi_metadata_type1.max_fall);
+        */
+
+        drmModeCreatePropertyBlob(ctx->connector->fd, &p->data, sizeof p->data, &p->blob_id);
+        drmModeAtomicReqPtr request = drmModeAtomicAlloc();
+        if (!request)
+            return;
+        drmModeAtomicAddProperty(request, ctx->connector->id, property_hdr_id, p->blob_id);
+        drmModeAtomicCommit(ctx->connector->fd, request, DRM_MODE_ATOMIC_ALLOW_MODESET, NULL);
```

Comment:
Hm, seems like I commented on the wrong PR. My comment on the previous one still applies here I think: https://github.com/mpv-player/mpv/pull/8456#discussion_r662936304

Reply:
If mpv doesn't support atomic KMS, there's a function to set a property without any of the atomic guarantees: `drmModeObjectSetProperty`.

Reply:
Additional comment in case I forget: I don't think `ALLOW_MODESET` is fine here. It allows the driver to glitch the screen, and you don't really want that to happen during playback.

Reply:
As mentioned in the older PR - one can trivially handle the suggestion. Simply use TEST_ONLY and on `drmModeAtomicCommit()` success, return the blob to the caller. The latter already does the final `drmModeAtomicCommit()` and can easy the property.